src/os/bluestore/bluestore_types.h

```diff
@@ -729,8 +730,11 @@ struct bluestore_blob_t {
     }
   }
 
-  int map(uint64_t x_off, uint64_t x_len,
-	   std::function<int(uint64_t,uint64_t)> f) const {
+  template<class F>
+  int map(uint64_t x_off, uint64_t x_len, F&& f) const {
+    static_assert(std::is_invocable_r_v<int, F, uint64_t, uint64_t>,
```

Comment:
please note `std::is_invocable_r_v<>` was introduced in C++17. while luminous is using C++14.
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/bluestore_types.h: In member function ‘int bluestore_blob_t::map(uint64_t, uint64_t, F&&) const’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/bluestore_types.h:735:19: error: ‘is_invocable_r_v’ is not a member of ‘std’
     static_assert(std::is_invocable_r_v<int, F, uint64_t, uint64_t>,
                   ^
```

Reply:
Yeah, the same with `std::is_invocable`. Just pushed a version with the asserts removed.